common/static/common/js/discussion/utils.js

```diff
@@ -444,19 +463,52 @@
             }
         };
 
-        DiscussionUtil.abbreviateHTML = function(html, minLength) {
+        DiscussionUtil.convertMath = function(element, alreadyEscaped) {
+            var elementContent = element.html();
+            if (alreadyEscaped){
+                // markdownWithHighlight requires unescaped text, so we must unescape it.
+                // safe-lint: disable=javascript-escape
+                elementContent = _.unescape(edx.HtmlUtils.HTML(elementContent));
+            }
```

Comment:
I've got a couple of questions about this code: 

1.  Since `element.html()` is coming out of the DOM, it needed to be safe before whatever it was made it into the DOM, otherwise it is an unsafe way to pass this data around.  
2. What is the case where this is double-escaped (which is what I presume you must mean by `alreadyEscaped`, and could we simply avoid double-escaping?
3. Because we are using `element.html()`, characters like `&` are going to be escaped already (i.e. `&amp;), so wouldn't everything need to be unescaped if all of this is going to be escaped again?
4. I'm not sure why the `disable` comment above is needed, nor why the call to `HtmlUtils.HTML` is needed inside the code to `_.unescape()`.

I don't know what the appropriate code here should be, because I'm unclear about what is unique about the single call to convertMath where alreadyEscaped is set to true.  Let me know if you need more help working this out.